hera_cal/nucal.py

```diff
@@ -621,4 +684,281 @@ def build_nucal_wgts(data_flags, data_nsamples, autocorrs, auto_flags, radial_re
+    use_spectral_filters = spectral_filters is not None
+
+    # Assert that the model components match the baselines in the radial_reds
+    for group in radial_reds:
+        assert group[0] in fg_model_comps, "fg_model_comps must contain a model component for each baseline in radial_reds."
+        if use_spectral_filters:
+            assert fg_model_comps[group[0]].shape[1:] == (spectral_filters.shape[-1], spatial_filters[group[0]].shape[-1]), (
+                f"The number of model components must match filter shapes."
+            )
+        else:
+            assert fg_model_comps[group[0]].shape[-1] == spatial_filters[group[0]].shape[-1], (
+                f"The number of model components in fg_model_comps must match the number of \
+                  eigenvectors in spatial_filters for baseline {group[0]}."
+            )
+
+    # Initialize a dictionary to hold the model components
+    model = {}
+
+    # Determine the einsum path to use for evaluating the model
+    einsum_path = "fm,fn,tmn->tf" if use_spectral_filters else "fm,tm->tf"
+
+    # Loop over the groups in radial_reds
+    for group in radial_reds:
+        for bl in group:
+            # Compute the model components for this baseline
+            if use_spectral_filters:
+                model[bl] = jnp.einsum(einsum_path, spectral_filters, spatial_filters[bl], fg_model_comps[group[0]])
+            else:
+                model[bl] = jnp.einsum(einsum_path, spatial_filters[bl], fg_model_comps[group[0]])
+
+    return model
+
+
+def fit_nucal_foreground_model(data, data_wgts, radial_reds, spatial_filters, spectral_filters=None, tol=1e-15,
+                               share_fg_model=False, return_model_comps=False, solver="lu_solve"):
+    """
+    Compute a foreground model for a set of radially redundant baselines. The model is computed by performing a linear
+    least-squares fit using a set of DPSS filters to model visibilities within a radially redundant group. If only spatial
+    filters are provided, the model is restricted to the spatial axis. If spectral filters are also provided, the model is fit
+    allowing spectral variation at a fixed spatial scale. The model components are return if return_model_comps is True. 
+    Otherwise, the model is evaluated using the model components and the spatial filters (and spectral filters if provided),
+    and the model visibilities are returned.
+    
+    Parameters:
+    -----------
+    data : dictionary, DataContainer
+        Data to be fit by u-model. This function assumes that visibilities redundantly-averaged
+        and will only use data from the first baseline in a spatially-redundant group.
+    data_wgts : dictionary, DataContainer
+        Weights associated with data
+    radial_reds : RadialRedundancy, list of lists of tuples
+        List of lists of redundant baseline tuples. Each list of redundant tuples represents a radial group of
```

Comment:
I'd clarify where you mean baseline redundancy and where you mean frequency redundancy

Reply:
Fixed this